component/espejote-templates/netpol-sync.jsonnet

```diff
@@ -0,0 +1,122 @@
+        namespace: namespace.metadata.name,
+      },
+    }),
+    esp.markForDelete({
+      apiVersion: resources.allowFromClusterNodes.apiVersion,
+      kind: resources.allowFromClusterNodes.kind,
+      metadata: {
+        name: resources.allowFromClusterNodes.metadata.name,
+        namespace: namespace.metadata.name,
+      },
+    }),
+  ];
+
+local applyBasePolicies(namespace) = [
+  resources.allowFromOtherWithSpec {
+    metadata+: {
+      namespace: namespace.metadata.name,
+    },
+  },
+] + if config.networkPlugin == 'cilium' then [
+  resources.allowFromNodesWithSpec {
+    metadata+: {
+      namespace: namespace.metadata.name,
+    },
+  },
+] else [];
+
+local applyDefaultPolicies(namespace) =
+  applyBasePolicies(namespace) + [
+    resources.allowFromSameWithSpec {
+      metadata+: {
+        namespace: namespace.metadata.name,
+      },
+    },
+  ];
+
+local reconcileNamespace(namespace) =
+  // Purge NetworkPolicies and CiliumNetworkPolicies if the namespace
+  // has the purgeNonBase label.
+  if isPurgeNonBase(namespace) then purgeNonBasePolicies(namespace)
+  // Purge NetworkPolicies and CiliumNetworkPolicies if the namespace
+  // is either in the ignoredNamespaces list or has the purgeDefaults label.
+  else if isPurgeDefaults(namespace) then purgeDefaultPolicies(namespace)
+  // Apply base NetworkPolicies and CiliumNetworkPolicies if the namespace
+  // has the baseDefaults label.
+  else if isApplyBase(namespace) then applyBasePolicies(namespace)
+  // Apply default NetworkPolicies and CiliumNetworkPolicies if the namespace
+  // nothing else applies.
+  else applyDefaultPolicies(namespace);
+
+// check if the object is getting deleted by checking if it has
+// `metadata.deletionTimestamp`.
+local inDelete(obj) = std.get(obj.metadata, 'deletionTimestamp', '') != '';
+
+// Do the thing
+if esp.triggerName() == 'namespace' then (
+  // Handle single namespace update on namespace trigger
+  local nsTrigger = esp.triggerData();
+  // nsTrigger can be null if we're called when the namespace is getting
+  // deleted. If it's not null, we still don't want to do anything when the
+  // namespace is getting deleted.
+  if nsTrigger != null && !inDelete(nsTrigger.resource) then
+    reconcileNamespace(nsTrigger.resource)
+) else if esp.triggerName() == 'netpol' || esp.triggerName() == 'ciliumpol' then (
+  // Handle single namespace update on netpol or ciliumpol trigger
+  local netpolTrigger = esp.triggerData();
+  if netpolTrigger != null && !inDelete(netpolTrigger.resource) then
+    local namespaceLabels = [
+      ns.metadata.labels
+      for ns in esp.context().namespace
+      if ns.metadata.name == netpolTrigger.resource.metadata.namespace
+    ];
+    local namespaceStub = {
+      metadata: {
+        labels: if std.length(namespaceLabels) == 1 then namespaceLabels[0] else {},
+        name: netpolTrigger.resource.metadata.namespace,
+      },
+    };
+    reconcileNamespace(namespaceStub)
```

Comment:
This works, when a NetworkPolicy is edited but fails when it gets deleted. 
I am inclined to remove processing netpol triggers, this means if one NetworkPolocy gets deleted it will trigger reconciliation of all namespaces.

Reply:
This is one of the cases where espejote maybe should also export raw trigger data.

Reply:
https://github.com/vshn/espejote/pull/93